internal/controller/drplacementcontrol.go

```diff
@@ -1573,13 +1573,39 @@ func (d *DRPCInstance) ensureVRGManifestWork(homeCluster string) error {
 	return d.mwu.UpdateVRGManifestWork(vrg, mw)
 }
 
+// equalClusterIDSlices compares 2 slices of clusterID strings and reports true if both contain the same clusterIDs.
+// This is not a generic routine that can be used for any pair of string slices for equality checks, as a case of,
+// - ["a", "b", "b"] compared to ["b", "a", "a"], would return true
+// It is used for clusterIDs with the limitation, because clusterIDs are based on UUIDs and collision has ignorable
+// probability
+func equalClusterIDSlices(a, b []string) bool {
+	if len(a) != len(b) {
+		return false
+	}
+
+	for _, v := range b {
+		if !slices.Contains(a, v) {
+			return false
+		}
+	}
+
+	// check the other way to ensure there are no extra elements in a that are not contained in b
+	for _, v := range a {
+		if !slices.Contains(b, v) {
+			return false
+		}
+	}
```

Comment:
(nit) This function works, but sorting the slices first and then checking for equality might be a cleaner and more efficient way to eliminate the extra loops.

Reply:
Sorting the slices was one thing I considered (and prototyped), that causes the slice to get updated, so we need to copy out the slices and then sort, rather than sorting the slices that are passed in.

Overall chose this instead of the sort, as either way the cost is the same, and we would always deal with 1 or 2 clusterIDs and no more, as peerClass is always between 2 clusters, even if there are more clusters (in the future) with a DRPolicy.

Reply:
Same comment as Benamar, this works but there are multiple places in Ramen where a slice should have been a set. In some places, we are already using "k8s.io/apimachinery/pkg/util/sets" to utilize the Equal/Union/Intersection functions. I think we should work on writing wrappers and use the same function everywhere. Can be done in a subsequent PR.

Reply:
I tested sets as well. In the set case, again it solves only the case of every element in A is in B, and needs a vice versa. So instead of adding Sets into the mix, which maybe good overall (and we need to update the sets package as the APIs currently in use are deprecated), retained this comparison instead.

Overall at least with sets the original slices remains untouched, so if required then I can change this.